src/System.Slices/src/System/Span.cs

```diff
@@ -335,7 +320,7 @@ public override bool Equals(object obj)
         /// gain: performance: no boundaries check (single operation) 
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private T GetItemWithoutBoundariesCheck(int index)
+        internal T GetItemWithoutBoundariesCheck(int index)
```

Comment:
I just wanted to get the best possible performance (boundaries check is 20% time of  the getter). Previously this method was private and used only for Enumerator, but I wanted to use it in exension methods, so I made it internal.

Is it possible to make RyuJit do boundaries check elimination for Slice as it does for Array?

On the other hand, the method name itself gives clear declaration of what it does, so nobody should misuse it.